TestConnectionError is flaky

```
=== RUN   TestConnectionError/CloseAt8ErrorAt7
--- FAIL: TestConnectionError (0.01s)
    --- PASS: TestConnectionError/CloseAt0ErrorAt0 (0.00s)
    --- PASS: TestConnectionError/CloseAt1ErrorAt1 (0.00s)
    --- PASS: TestConnectionError/CloseAt2ErrorAt2 (0.00s)
    --- PASS: TestConnectionError/CloseAt3ErrorAt3 (0.00s)
    --- FAIL: TestConnectionError/CloseAt4ErrorAt4 (0.00s)
        packet_test.go:305: Unexpected error: read/write on closed transport
    --- PASS: TestConnectionError/CloseAt5ErrorAt5 (0.00s)
    --- PASS: TestConnectionError/CloseAt6ErrorAt6 (0.00s)
    --- PASS: TestConnectionError/CloseAt7ErrorAt6 (0.00s)
    --- PASS: TestConnectionError/CloseAt8ErrorAt7 (0.00s)

--- a/packet_test.go
+++ b/packet_test.go
@@ -278,6 +278,7 @@ func TestConnectionError(t *testing.T) {
 			go func() {
 				defer cb.Close()
 				for i, resp := range resps {
+					time.Sleep(10 * time.Millisecond)
 					if i == c.closeAt {
 						return
 					}
@@ -302,7 +303,7 @@ func TestConnectionError(t *testing.T) {
 					break
 				} else {
 					if err != nil {
-						t.Errorf("Unexpected error: %v", err)
+						t.Errorf("Unexpected error at %d: %v", i, err)
 					}
 				}
 			}

